src/SFA.DAS.AssessorService.Web/Controllers/CertificateBaseController.cs

```diff
@@ -55,9 +55,17 @@ protected async Task<IActionResult> LoadViewModel<T>(string view) where T : Cert
 
             Logger.LogInformation($"Got View Model of type {typeof(T).Name} requested by {username}");
 
+            if (!returnView && viewModel.GetType() == typeof(CertificateNamesViewModel)) { SetNamePropertyAndReturnView(viewModel as CertificateNamesViewModel); }
```

Comment:
This is a red flag - a base class needing to know the type of one of its derived classes